packages/playground/website/src/components/ensure-playground-site/ensure-playground-site-is-selected.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+	setActiveSite,
+	useAppDispatch,
+	useAppSelector,
+} from '../../lib/state/redux/store';
+import { randomSiteName } from '../../lib/state/redux/random-site-name';
+import { PlaygroundRoute, redirectTo } from '../../lib/state/url/router';
+import { logger } from '@php-wasm/logger';
+
+/**
+ * Ensures the redux store always has an activeSite value.
+ *
+ * It has two routing modes:
+ * * When `site-slug` is provided, it load an existing site
+ * * When `site-slug` is missing, it creates a new site using the Query API and Blueprint API
+ *   data sourced from the current URL.
+ */
+export function EnsurePlaygroundSiteIsSelected({
+	children,
+}: {
+	children: React.ReactNode;
+}) {
+	const siteListingStatus = useAppSelector(
+		(state) => state.sites.loadingState
+	);
+	const sites = useAppSelector((state) => state.sites.entities);
+	const dispatch = useAppDispatch();
+	const url = useCurrentUrl();
+	const requestedSiteSlug = url.searchParams.get('site-slug');
+	const requestedSiteObject = useAppSelector((state) =>
+		selectSiteBySlug(state, requestedSiteSlug!)
+	);
+
+	useEffect(() => {
+		opfsSiteStorage?.list().then(
+			(sites) => dispatch(siteListingLoaded(sites)),
+			(error) => {
+				logger.error('Error loading sites:', error);
+				dispatch(siteListingLoaded([]));
+			}
+		);
+	}, [dispatch]);
+
+	useEffect(() => {
+		async function ensureSiteIsSelected() {
+			// If the site slug is provided, try to load the site.
+			if (requestedSiteSlug) {
+				// Wait until the site listing is loaded
+				if (siteListingStatus !== 'loaded') {
+					return;
+				}
+
+				// If the site does not exist, redirect to a new temporary site.
+				if (!requestedSiteObject) {
+					// @TODO: Notification: 'The requested site was not found. Redirecting to a new temporary site.'
+					logger.log(
+						'The requested site was not found. Redirecting to a new temporary site.'
+					);
+					redirectTo(PlaygroundRoute.newTemporarySite());
+					return;
+				}
+				dispatch(setActiveSite(requestedSiteSlug));
+				return;
+			}
+
+			// If the site slug is missing, create a new temporary site.
+			// Lean on the Query API parameters and the Blueprint API to
+			// create the new site.
+			const url = new URL(window.location.href);
+			const blueprint = await resolveBlueprintFromURL(url);
+			const siteNameFromUrl = url.searchParams.get('name')?.trim();
+			const urlParams = {
+				searchParams: Object.fromEntries(url.searchParams.entries()),
+				hash: url.hash,
+			};
+			const newSiteInfo = await createNewSiteInfo({
+				metadata: {
+					name: siteNameFromUrl || undefined,
+					originalBlueprint: blueprint,
+				},
+				originalUrlParams: urlParams,
+			});
+
+			// Check if there's an existing site that matches the requested
+			// specification exactly.
+			const existingSite = Object.values(sites).find(
+				(site) =>
+					JSON.stringify(site.originalUrlParams) ===
+					JSON.stringify(urlParams)
+			);
+			if (existingSite) {
```

Comment:
I really like this thoughtfulness but wonder whether we will regret this being implicit behavior. I guess we will see because, at first glance, it feels like a nice, thoughtful feature.

Reply:
It is the implicit behavior today, I just don't want to break that.

Reply:
> It is the implicit behavior today, I just don't want to break that.

That seems good. Thanks